### ISSUE TITLE
.net 6 WebAPI Template requires program.cs restructuring to add a Log Provider class

### ISSUE THREAD

**Issue by yogigrantz** [dotnet/aspnetcore#48575]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

When creating a brand new WebAPI solution, Visual Studio uses a WebAPI template that produces this code in program.cs, right out of the box, along with WeatherForcast.cs and WeatherForecastController:
```

var builder = WebApplication.CreateBuilder(args);

// Add services to the container.

builder.Services.AddControllers();

var app = builder.Build();

// Configure the HTTP request pipeline.

app.UseAuthorization();

app.MapControllers();

app.Run();

```

This works fine until I need to add my custom Logger Provider class that inherits from ILogger. In order for me to add my custom Logger, I would have to modify the entire code above, to something like this: 
```

using CCLogProvider;

var host = Host.CreateDefaultBuilder(args)
                .ConfigureWebHostDefaults(webBuilder =>
                {
                    using (CCFileLoggerProvider logger = new CCFileLoggerProvider(LogLevel.Information, @"C:\Temp\Log", "testLog.txt", 3, 10000))
                    {
                        webBuilder.ConfigureLogging(loggingBuilder =>
                        {
                            loggingBuilder.ClearProviders();
                            loggingBuilder.AddProvider(logger);
                        });
                    }

                    webBuilder.ConfigureServices((hostingContext, services) =>
                     {
                         services.AddControllers();
                     });

                    webBuilder.Configure(app =>
                    {
                        app.UseRouting();
                        app.UseAuthorization();
                        app.UseEndpoints(endpoints =>
                        {
                            endpoints.MapControllers();
                        });
                    });
                })
                .Build();

    host.Run();


  // Remember to remove appsettings.json to log everything


```

That looks nothing like the original code. We are using Host.CreateDefaultBuilder approach instead of the generated WebApplication.CreateBuilder approach

### Describe the solution you'd like

Maybe if we can make the code generator to generates this code whenever we create a brand new WebAPI application:

```
var host = Host.CreateDefaultBuilder(args)
            .ConfigureWebHostDefaults(webBuilder =>
            {
                webBuilder.ConfigureServices((hostingContext, services) =>
                {
                    services.AddControllers();
                });

                webBuilder.Configure(app =>
                {
                    app.UseRouting();
                    app.UseEndpoints(endpoints =>
                    {
                        endpoints.MapControllers();
                    });
                });
            })
            .Build();

    host.Run();

```

We could easily add new DI objects such as custom log provider class. 

Just a thought, I'm open to any suggestions, thank you

**Comment by BrennanConroy** [comment]
It should be a single line (or double if you want to clear providers)
https://learn.microsoft.com/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-7.0#add-logging-providers

**Comment by yogigrantz** [comment]
@BrennanConroy  : Are you referring to this code?:

```
// Configure JSON logging to the console.
builder.Logging.AddJsonConsole();

```

Because that code would not allow custom Logger class. Could you please clarify?

**Comment by yogigrantz** [comment]
My other question is, why can't we use Host.CreateDefaultBuilder by default instead of WebApplication.CreateBuilder. To me Host.CreateDefaultBuilder seems to be more expandable

**Comment by BrennanConroy** [comment]
```
builder.Logging.ClearProviders();
builder.Logging.AddProvider(logger);
```

**Comment by yogigrantz** [comment]
@BrennanConroy  : that works, thank you!

**Comment by yogigrantz** [comment]
builder.Logging.AddProvider(logger); // works

**Comment by davidfowl** [comment]
> This works fine until I need to add my custom Logger Provider class that inherits from ILogger. In order for me to add my custom Logger, I would have to modify the entire code above, to something like this:

How did you come to this conclusion initially?

**Comment by yogigrantz** [comment]
@davidfowl  : Sorry, I got that from ChatGPT. I asked ChatGPT how to use custom Logger in WebAPI application, and it gave me that approach to use  
```

var host = Host.CreateDefaultBuilder(args)
                .ConfigureWebHostDefaults(webBuilder => .... 
```

which looks different from the generated code that uses WebApplication.CreateBuilder. So I thought I had to modify the structure. And I did. But, after @BrennanConroy  clarified that it can simply be added with one liner builder.Logging.AddProvider(logger), I went back and modified my WebAPI apps to the original generated style with that one liner, along with other things, i.e: swagger and cors
